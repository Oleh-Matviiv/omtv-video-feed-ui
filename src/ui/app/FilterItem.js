import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

export const Div = styled.div.attrs({
  className: 'filter-item',
})`
  padding: .5rem;
  margin: 1rem .5rem 0;
  border: 1px solid #ccc;
  cursor: pointer;
  background: ${props => props.selected ? '#eee' : 'none'};
`;

const FilterItem = ({filter, handleFilterChange, selected}) => (
  <Div
    onClick={handleFilterChange}
    value={filter}
    selected={selected}
  >
    {filter}
  </Div>
);
FilterItem.propTypes = {
  filter: PropTypes.oneOf([
    'any', 'facebook', 'youtube', 'url',
  ]).isRequired,
  handleFilterChange: PropTypes.func.isRequired,
  selected: PropTypes.bool.isRequired,
};

export default FilterItem;
