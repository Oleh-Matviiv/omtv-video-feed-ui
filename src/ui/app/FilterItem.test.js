import React from 'react';
import { mount, shallow } from 'enzyme';
import 'jest-styled-components';

import FilterItem, { Div } from './FilterItem';

describe('snapshots', () => {
  it('FilterItem', () => {
    const wrapper = shallow(
      <FilterItem
        filter="facebook"
        handleFilterChange={jest.fn()}
        selected={false}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Div', () => {
    const wrapper = mount(
      <Div selected={false}>facebook</Div>
    );

    expect(wrapper).toMatchSnapshot();
  });
  it('Div selected', () => {
    const wrapper = mount(
      <Div selected={true}>any</Div>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
