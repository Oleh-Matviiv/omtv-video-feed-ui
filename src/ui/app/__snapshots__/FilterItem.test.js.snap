// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Div 1`] = `
.c0 {
  padding: .5rem;
  margin: 1rem .5rem 0;
  border: 1px solid #ccc;
  cursor: pointer;
  background: none;
}

<styled.div
  selected={false}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "filter-item",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  padding: .5rem;
  margin: 1rem .5rem 0;
  border: 1px solid #ccc;
  cursor: pointer;
  background: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    selected={false}
  >
    <div
      className="filter-item c0"
      selected={false}
    >
      facebook
    </div>
  </StyledComponent>
</styled.div>
`;

exports[`snapshots Div selected 1`] = `
.c0 {
  padding: .5rem;
  margin: 1rem .5rem 0;
  border: 1px solid #ccc;
  cursor: pointer;
  background: #eee;
}

<styled.div
  selected={true}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "className": "filter-item",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  padding: .5rem;
  margin: 1rem .5rem 0;
  border: 1px solid #ccc;
  cursor: pointer;
  background: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    selected={true}
  >
    <div
      className="filter-item c0"
      selected={true}
    >
      any
    </div>
  </StyledComponent>
</styled.div>
`;

exports[`snapshots FilterItem 1`] = `
<styled.div
  onClick={[MockFunction]}
  selected={false}
  value="facebook"
>
  facebook
</styled.div>
`;
